pkg/logql/engine.go

```diff
@@ -214,8 +214,17 @@ func (q *query) resultLength(res promql_parser.Value) int {
 func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "query.Exec")
 	defer sp.Finish()
-	log := spanlogger.FromContext(ctx)
-	defer log.Finish()
+	spLogger := spanlogger.FromContext(ctx)
+	defer spLogger.Finish()
+
+	level.Debug(spLogger).Log(
```

Comment:
that would log a new line for every exec (which tbh isn't that bad). But what do you think of adding a new event to the span and adding those params to it?

Reply:
Done. Looking at the opentracing docs, it's not super clear to me whether LogKV logs to stdout or not.
Even if it does, looks like it uses some buffered logging approach which should make it way faster ([see][1]).

I see this being used elsewhere within Loki. For example in the Distributor _push_ handler:
https://github.com/grafana/loki/blob/24d426ce4400d3197eef6698c1c058e12db652bb/pkg/distributor/distributor.go#L298

[1]: https://github.com/grafana/loki/blob/d765921735c971d5323f6950311c324f421ef7a9/vendor/github.com/uber/jaeger-client-go/span.go#L266